File: web/src/layout/ExampleContentXml.ts
//
// ExampleContentXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'butterfly/dist/views/html'
import { customViewSetDelegate } from 'butterfly/dist/views/CustomView'
import htmlForDefault from './example_content.html'
//! Declares com.lightningkite.butterflytemplate.layouts.ExampleContentXml
export class ExampleContentXml {
    xmlRoot!: HTMLElement;
    scrollView!: HTMLDivElement;
    exampleContentNumber!: HTMLElement;
    exampleContentIncrement!: HTMLButtonElement;
    chainedNumber!: HTMLElement;
    chainedIncrement!: HTMLButtonElement;
    scrollToTop!: HTMLButtonElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.scrollView = getViewById<HTMLDivElement>(view, "scrollView");
        this.exampleContentNumber = getViewById<HTMLElement>(view, "exampleContentNumber");
        this.exampleContentIncrement = getViewById<HTMLButtonElement>(view, "exampleContentIncrement");
        this.chainedNumber = getViewById<HTMLElement>(view, "chainedNumber");
        this.chainedIncrement = getViewById<HTMLButtonElement>(view, "chainedIncrement");
        this.scrollToTop = getViewById<HTMLButtonElement>(view, "scrollToTop");
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/ExampleContentXml.kt
//
// ExampleContentXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class ExampleContentXml {

    lateinit var scrollView: ScrollView
    lateinit var exampleContentNumber: TextView
    lateinit var exampleContentIncrement: Button
    lateinit var chainedNumber: TextView
    lateinit var chainedIncrement: Button
    lateinit var scrollToTop: Button
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.example_content, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        scrollView = view.findViewById<ScrollView>(R.id.scrollView)
        exampleContentNumber = view.findViewById<TextView>(R.id.exampleContentNumber)
        exampleContentIncrement = view.findViewById<Button>(R.id.exampleContentIncrement)
        chainedNumber = view.findViewById<TextView>(R.id.chainedNumber)
        chainedIncrement = view.findViewById<Button>(R.id.chainedIncrement)
        scrollToTop = view.findViewById<Button>(R.id.scrollToTop)
        
        
        
        return view
    }
}